src/jailer/src/cgroup.rs

```diff
@@ -28,18 +22,19 @@ struct CgroupMountPoint {
 
 // Allows creation of cgroups on the system for both versions
 #[derive(Debug)]
-pub struct CgroupBuilder {
+pub struct CgroupBuilder<'a> {
     version: u8,
     hierarchies: HashMap<String, PathBuf>,
     mount_points: Vec<CgroupMountPoint>,
+    proc_mounts_path: &'a str,
 }
```

Comment:
This may use more memory than before.

It would be better to do:
```suggestion
#[derive(Debug)]
pub struct CgroupBuilder {
    version: u8,
    hierarchies: HashMap<String, PathBuf>,
    mount_points: Vec<CgroupMountPoint>,
    proc_mounts_path: &'static str,
}
```

Reply:
@JonathanWoollett-Light 
Thank you for your suggestion. However, since the field requires dynamic values. 
Because it's necessary to isolate the environment between multiple threads running tests.
If static lifetime were used, multiple threads running tests would access the same directory, potentially causing fail tests.

Reply:
Although only for testing, dynamic values are used in the following places:

https://github.com/firecracker-microvm/firecracker/blob/14f8f38b9819514956eb5b9103ec7cd9cccdd6a1/src/jailer/src/cgroup.rs#L477-L479

https://github.com/firecracker-microvm/firecracker/blob/14f8f38b9819514956eb5b9103ec7cd9cccdd6a1/src/jailer/src/cgroup.rs#L608

How about changing the type of proc_mounts_path to Cow<'static, str>? With this change, the production code can accept the constant PROC_MOUNTS, while the test code can dynamically generate a String. What do you think about this approach?